tests/integrationv2/utils.py

```diff
@@ -51,6 +51,28 @@
     return str(item)
 
 
+def expected_signature(protocol, signature):
+    if protocol < Protocols.TLS12:
+        if signature.sig_type == 'ECDSA':
+            signature = Signatures.ECDSA_SHA1
+        else:
+            signature = Signatures.MD5_SHA1
+    else:
+        signature = signature
```

Comment:
This check seems legitimate. `signature` doesn't need to be set to itself. In general, it seems weird to override the `signature` argument and then return it. It could instead just return the expected signature in each branch.